lib/type/boolean.ex

```diff
@@ -1,15 +1,79 @@
 defmodule Litmus.Type.Boolean do
-  @moduledoc """
-  Schema and validation for Boolean data type.
-  """
+  @moduledoc false
 
-  defstruct truthy: [true, "true"],
-            falsy: [false, "false"],
+  alias Litmus.Required
+
+  @truthy_default [true, "true"]
+  @falsy_default [false, "false"]
+
+  defstruct truthy: @truthy_default,
+            falsy: @falsy_default,
             required: false
 
   @type t :: %__MODULE__{
           truthy: [term],
           falsy: [term],
           required: boolean
         }
+
+  @spec validate_field(t, binary, map) :: {:ok, map} | {:error, binary}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- truthy_falsy_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec check_boolean_values(term, [term], [term]) :: {:ok, map} | nil
+  def check_boolean_values(initial_value, additional_values, default_values)
+      when is_binary(initial_value) do
+    allowed_values = Enum.uniq(additional_values ++ default_values)
+
+    allowed_values =
+      Enum.map(allowed_values, fn item ->
+        if is_binary(item) do
+          String.downcase(item)
+        end
+      end)
+
+    if String.downcase(initial_value) in allowed_values do
```

Comment:
You can make this one line:

```elixir
if String.downcase(initial_value) in allowed_values, do: true
```

By default, functions return nil, so if you leave out the `else` case the behavior will be the same.